frontend/src/component/AddVisitCountry.tsx

```diff
@@ -0,0 +1,56 @@
+import {Country} from "../model/Country";
+import {ChangeEvent, useState} from "react";
+import VisitCountryCard from "./VisitCountryCard";
+import "./AddVisitCountry.css";
+import {User} from "../model/User";
+import {useNavigate} from "react-router-dom";
+import {updateUser} from "../apiCalls";
+
+type AddVisitCountryProps = {
+    countries: Country[],
+    loggedInUser: User
+}
+export default function AddVisitCountry(props: AddVisitCountryProps) {
+    const [searchQuery, setSearchQuery] = useState("")
+    const navigate = useNavigate()
+
+    const filteredCountries = props.countries.filter(country => {
+        return country.name.toLowerCase().includes(searchQuery.toLowerCase())
+            || country.threeLetterCode.toLowerCase().includes(searchQuery.toLowerCase())
+            || country.flag.toLowerCase().includes(searchQuery.toLowerCase())
+    })
+
+    function handleSearchText(event: ChangeEvent<HTMLInputElement>) {
+        setSearchQuery(event.target.value)
+    }
+
+
+    function addCountryToUser(country: Country) {
+        if (!props.loggedInUser.visitedCountries) {
+            props.loggedInUser.visitedCountries = []
+        }
+
+        props.loggedInUser.visitedCountries.push(country)
+
+        updateUser(props.loggedInUser.id, props.loggedInUser)
+            .then(() => {
+                navigate("/overview")
+            })
+            .catch(console.error)
+    }
+
+    return (
+        <div>
+            <div>
+                <input type={"search"} onChange={handleSearchText} placeholder={"Search for a country.."}/>
+            </div>
+            <div className={"card-display"}>
+                {filteredCountries.length < 5 && filteredCountries.map((value) => {
```

Comment:
Why are you only displaying your list of countries when there a less than 5? 🤔

Reply:
Otherwise its very Hugh and looks ugly

Reply:
Maybe you could only display the first 5 even if there are more?